capsules/src/i2c_master_slave_driver.rs

```diff
@@ -90,41 +90,36 @@ impl hil::i2c::I2CHwMasterClient for I2CMasterSlaveDriver<'_> {
                 self.master_buffer.replace(buffer);
 
                 self.app.map(|app| {
-                    app.callback.map(|mut cb| {
-                        cb.schedule(0, err as usize, 0);
-                    });
+                    app.callback.schedule(0, err as usize, 0);
                 });
             }
 
             MasterAction::Read(read_len) => {
                 self.app.map(|app| {
-                    app.master_rx_buffer.as_mut().map(move |app_buffer| {
+                    app.master_rx_buffer.mut_map_or(0, move |app_buffer| {
```

Comment:
```suggestion
                    app.master_rx_buffer.mut_map(move |app_buffer| {
```

The return value isn't used here -- we should either add a `mut_map` method, or do something meaningful here in the case where the map fails.

Reply:
Yes -- this is an old capsule and has poor error handling.